doc/book/zend.authentication.adapter.dbtable.md

```diff
@@ -0,0 +1,304 @@
+# Database Table Authentication
+
+> **note**
+>
+> `Zend\Authentication\Adapter\DbTable` has been deprecated, as its responsibilities have been
+splitted off into `Zend\Authentication\Adapter\DbTable\CallbackCheck` and
+`Zend\Authentication\Adapter\DbTable\CredentialTreatmentAdapter`. Use
+`Zend\Authentication\Adapter\DbTable\CredentialTreatmentAdapter` instead of
+`Zend\Authentication\Adapter\DbTable`.
+
+## Introduction
+
+`Zend\Authentication\Adapter\DbTable` provides the ability to authenticate against credentials
+stored in a database table. Because `Zend\Authentication\Adapter\DbTable` requires an instance of
+`Zend\Db\Adapter\Adapter` to be passed to its constructor, each instance is bound to a particular
+database connection. Other configuration options may be set through the constructor and through
+instance methods, one for each option.
+
+The available configuration options include:
+
+-   **tableName**: This is the name of the database table that contains the authentication
+credentials, and against which the database authentication query is performed.
+-   **identityColumn**: This is the name of the database table column used to represent the
+identity. The identity column must contain unique values, such as a username or e-mail address.
+-   **credentialColumn**: This is the name of the database table column used to represent the
+credential. Under a simple identity and password authentication scheme, the credential value
+corresponds to the password. See also the `credentialTreatment` option.
+-   **credentialTreatment**: In many cases, passwords and other sensitive data are encrypted,
+hashed, encoded, obscured, salted or otherwise treated through some function or algorithm. By
+specifying a parameterized treatment string with this method, such as '`MD5(?)`' or '`PASSWORD(?)`',
+a developer may apply such arbitrary *SQL* upon input credential data. Since these functions are
+specific to the underlying *RDBMS*, check the database manual for the availability of such functions
+for your database system.
+
+## Basic Usage
+
+As explained in the introduction, the `Zend\Authentication\Adapter\DbTable` constructor requires an
+instance of `Zend\Db\Adapter\Adapter` that serves as the database connection to which the
+authentication adapter instance is bound. First, the database connection should be created.
+
+The following code creates an adapter for an in-memory database, creates a simple table schema, and
+inserts a row against which we can perform an authentication query later. This example requires the
+*PDO* SQLite extension to be available:
+
+```php
+use Zend\Db\Adapter\Adapter as DbAdapter;
+
+// Create a SQLite database connection
+$dbAdapter = new DbAdapter(array(
+                'driver' => 'Pdo_Sqlite',
+                'database' => 'path/to/sqlite.db'
+            ));
```

Comment:
Since we have the opportunity, we should likely reformat these to follow PSR-1/2.